tests/integration/go_ethereum/test_goethereum_http.py

```diff
@@ -30,18 +30,36 @@ def endpoint_uri(rpc_port):
 
 
 @pytest.fixture(scope='module')
-def geth_command_arguments(geth_binary, datadir, rpc_port):
-    return (
-        geth_binary,
-        '--datadir', str(datadir),
-        '--shh',
-        '--nodiscover',
-        '--fakepow',
-        '--rpc',
-        '--rpcport', rpc_port,
-        '--rpcapi', 'admin,db,eth,net,web3,personal,shh,web3',
-        '--ipcdisable',
-    )
+def geth_command_arguments(rpc_port,
+                           base_geth_command_arguments,
+                           get_geth_version):
+
+    if get_geth_version.major == 1:
+        if get_geth_version.minor == 9:
+            return (
+                base_geth_command_arguments +
+                (
+                    '--rpc',
+                    '--rpcport', rpc_port,
+                    '--rpcapi', 'admin,db,eth,net,web3,personal,shh,miner',
+                    '--ipcdisable',
+                    '--allow-insecure-unlock',
+                )
+            )
+        elif get_geth_version.minor == 8 or get_geth_version.minor == 7:
+            return (
+                base_geth_command_arguments +
+                (
+                    '--rpc',
+                    '--rpcport', rpc_port,
+                    '--rpcapi', 'admin,db,eth,net,web3,personal,shh,miner',
+                    '--ipcdisable',
+                )
+            )
+        else:
+            assert False, "Unsupported geth version"
+    else:
```

Comment:
I think you can cleanup this function by using a `yield` based approach.

```python
def get_command_args(...):
    yield from base_geth_command_arguments
    if version.major == 1:
        yield from (
            '--rpc',
            '--rpcport', rpc_port,
            '--rpcapi', 'admin,db,eth,net,web3,personal,shh,miner',
            '--ipcdisable',
        )
        if version.minor == 9:
            yield "--allow-insecure-unlock"
```

The code above doesn't contain the "unsupported geth version" checks but should be demonstrative of the concept.